File: ts/project/.sdk/tm/zig/utility/jsonparse.zig
// JSON text -> voxgig struct Value, via std.json at the boundary then
// fromStdJson (keeps the SDK runtime free of a hand-rolled parser). Used by
// the default live transport; the offline test transport builds Values
// directly.

const std = @import("std");
const vs = @import("voxgig-struct");
const h = @import("../core/helpers.zig");

pub fn json_parse(src: []const u8) !vs.JsonValue {
    const parsed = try std.json.parseFromSlice(std.json.Value, h.A(), src, .{});
    return vs.fromStdJson(h.A(), parsed.value);
}

File: ts/project/.sdk/tm/c/utility/jsonparse.c
// JSON parse bridge — wraps the vendored struct JSON parser.

#include "sdk.h"

#include <string.h>

voxgig_value* json_parse(const char* text) {
  if (!text) return voxgig_new_undef();
  voxgig_value* v = voxgig_parse_json(text, strlen(text));
  return v ? v : voxgig_new_undef();
}
